### PR TITLE
Added UserParam class for user parameter specification

### PR DESCRIPTION
**Pull request by digicosmos86**
Note: To avoid conflict with `src/jhssm/param.py`, I am naming the directory `src/hssm/param_refactor` for now. Once all integration is tested, I will rename this directory

What was done in this PR:

1. Added a few methods in `prior.py` to prepare for model output
2. Added a UserParam (which will eventually be exported as `hssm.Param`) for users to specify priors and store user specification

### CODE REVIEW

**Review by AlexanderFengler — APPROVED** [review]
Prob warrants a comprehensive tutorial on prior setting?
Looks good otherwise.

File: tests/test_prior.py
Comment (cpaniaguam) [inline tests/test_prior.py:1]: Why not use the truthy/falsy nature of the values in many of these checks? Makes for more consice code.

**Review by cpaniaguam — CHANGES_REQUESTED** [review]
Looks good. I pointed out a couple of things I would change but might not be needed perhaps due to conventions that I'm not aware of yet.

**Comment by AlexanderFengler** [comment]
@digicosmos86 is this on hold or pushed through #590 at this point?

**Comment by AlexanderFengler** [comment]
From what I can see this is redundant now.

**Pull request closed without merging**